configuration/download_cache.rst

```diff
@@ -0,0 +1,92 @@
+  This includes files like *conanfile.py*, but also the zipped artifacts like *conan_package.tgz* or *conan_sources.tgz*.
+- The downloads done by users with the ``tools.download()`` or ``tools.get()`` helpers, as long as they provide a checksum (md5, sha1, etc.). If
+  a checksum is not provided, even if the download cache is enabled, the download will be always executed and the files will not be cached.
+
+The cache uses the URL of the download, appending the checksum when provided. In the api V2, it is not necessary to append the checksum, because
+the URL itself already encodes the recipe revision and/or the package revisions, which are already checksums of the recipe and package respectively.
+
+.. warning::
+
+    The download cache will not be able to correctly cache artifacts with revisions enabled if those artifacts are created and uploaded repeatedly
+    in a client without revisions, because that will keep overwriting the revision "0".
+
+Activating/deactivating the download cache
+------------------------------------------
+
+The download cache is activated and configured in the :ref:`conan_conf` like this:
+
+.. code-block:: text
+
+    [storage]
+    download_cache=/path/to/my/cache
+
+It can be defined from the command like as well:
+
+.. code-block:: bash
+
+    $ conan config set storage.download_cache="/path/to/my/cache"
+    # Display it
+    $ conan config get storage.download_cache
+
+
+And, as the *conan.conf* is part of the configuration, you can also put a common *conan.conf* file in a git repo or zip file and use
+the :ref:`conan_config_install` command to automatically install it in Conan clients.
+
+
+To deactivate the download cache, you can remove the entry ``download_cache`` from the *conan.conf* file directly editing it, or with the command:
+
+.. code-block:: bash
+
+    $ conan config rm storage.download_cache
+
+
+Concurrency, multiple caches and CI
+-----------------------------------
+
+The downloads cache implements exclusive locks for concurrency, so it can be shared among different concurrent Conan instances.
+This is a typical scenario in CI servers, in which each job uses a different Conan package cache (defined by ``CONAN_USER_HOME`` environment
+variable). Every different Conan instance could configure its download cache to share the same storage. The download cache implements interprocess
+exclusive locks, so only 1 process will access at a time to a given cached artifact. If other processes needs the same artifact, they will wait
+until it is released, avoiding multiple downloads of the same file, even if they were requested almost simultaneously.
+
+For Continuous Integration processes, it is recommended to have a different Conan package cache (``CONAN_USER_HOME``) for each job, in most of the cases,
+because the Conan package cache is not concurrent, and it might also have old dependencies, stale packages, etc. It is better to run CI jobs in a clean
+environment.
```

Comment:
It is ok to mention here that the _Download cache_ is implemented thread and process safe, but I'd move the things related to CI to the place in the docs where we are talking about CI integrations. There people can read all the recommendations together: download cache, do not share user_home, link to the Artifactory-Jenkins plugin, some example jenkinsfile,...

Reply:
The https://docs.conan.io/en/latest/integrations/ci.html is not the right place for this. That section is describing other technologies, and ways to integrate, not "good practices", there is not a general "CI" section yet. We should probably add one in the docs, but lets leave this here until them, as this is a good rationale of the existence of this download cache.

Reply:
Yes, that's not the place, maybe we need one.